prisma/schema.prisma

```
@@ -100,6 +100,8 @@ model Event {
   summary       String          @default("")
   enrol         String          @default("")
   content       String          @default("")
+  enrolKey      String          @default("")
```

```suggestion
  enrolKey      String          @default(uuid())
```

---

perhaps set this by default to a random string / uuid?, and the instructor key as well

---

I guess this would affect the migration as well

---

Ironically, by changing the enroldialog to use a form, it means you can't submit a blank string anyway (if the enrolkey is blank it wont match the undefined form submission for submitting a blank string) but I will make this change.

I had initially thought about hashing the key but this makes the edit event page significantly more complex as I would need a separate method of submitting the keys as you cant have them populated by default and therefore submitting the changes without setting a key would change the key. 

I figured since that actually without the hashing there is no real layer of security; the keys are exposed in the browser memory because the Event object is there so I just left it, but yeah this change is fine.

---

ahhh yes, probably with the form this doesn't really add much :) I agree with your thoughts re hashing, probably not worth implementing this as we don't need high security (at least I don't think we do.....)